### PR TITLE
Add Bookmark Folder link duplicates Bookmarks that already exist

### PR DESCRIPTION
**Pull request by flesher1**
Fixes issue where the add listing links will add multiple of the same
listing to existing folder.
This has become more apparent due to the new undelete feature.
Fixes functionality when undo is clicked from in-HUD popup, the system
notification is removed for the user.

Closes# AMLNG-887

**Description**
If a user has a bookmark folder (Animals) and goes to the URL to add/share a bookmark folder of the same name, with the same bookmarks, those bookmarks are duplicated, rather than just adding new ones. The duplicates are unnecessary, and can cause issues with page reloading.

**Acceptance Criteria**
A shared bookmark folder link will not duplicate bookmarks the user already has in a folder of the same name.

**How to Test**
Pull this branch. 
Go into a folder and copy the share address. 
Pasting the share address into your browser and going to it, the Hud should not change. 
Delete a listing from the folder and redo the previous steps, the folder should be back to the way it was. 

Delete the folder, Click undo. There should be no new notifications after reloading the site.
Delete the folder again. Do not click undo. There should be a notification after reloading the site.

### CODE REVIEW

File: app/js/actions/Library.js [hunk 1]
```diff
@@ -105,5 +105,11 @@ module.exports = Reflux.createActions([
     /**
      * Indicate that no specific folder is currently being viewed
      */
-    'stopViewingFolder'
+    'stopViewingFolder',
+
+    /**
+     * Remove notification after a foder is restored via undo button
```
Comment (blynch11p) [inline app/js/actions/Library.js]: `foder` typo

File: app/js/api/Library.js [hunk 2]
```diff
@@ -93,18 +93,31 @@ module.exports.LibraryApi = {
       });
     },
 
+    removeFolderNotification: function(notificaitonId){
```
Comment (blynch11p) [inline app/js/api/Library.js]: `notificaitonId` typo

File: app/js/api/Library.js [hunk 3]
```diff
@@ -93,18 +93,31 @@ module.exports.LibraryApi = {
...
+        return $.ajax({
+            url: `${API_URL}/api/notification/` + notificaitonId + '/',
```
Comment (blynch11p) [inline app/js/api/Library.js]: `notificaitonId` carried to here.

File: app/js/components/folder/Add.jsx [hunk 4]
```diff
@@ -6,16 +6,39 @@ var Reflux = require('reflux');
...
-    componentDidMount: function() {
-      var folderName = decodeURIComponent(this.props.params.name);
-      var ids = this.props.params.ids;
-      var idList = ids ?ids.split(',').map(Number) : [];
+            if (this.props.params.ids === null){
+                history.pushState(null, '', HUD_URL);
+            }
+            var folderName = decodeURIComponent(this.props.params.name);
+            var ids = this.props.params.ids;
+            var idList = ids ?ids.split(',').map(Number) : [];
```
Comment (blynch11p) [inline app/js/components/folder/Add.jsx]: add `<space>` after `?` for readability

File: app/js/components/library/FolderTile.jsx [hunk 5]
```diff
@@ -70,15 +71,14 @@ var FolderTile = React.createClass({
...
-        //if(window.confirm("Delete folder?")){
-            LibraryActions.deleteFolder(folder);
-            LibraryActions.restoreFolderNotification({
-              folder: folder.name,
-              peer: CurrentProfileStore.profile.username,
-              message: ''
-            });
-            deletedFn(folder);
-        //}
+        LibraryActions.deleteFolder(folder);
```
Comment (irizvifearless) [inline app/js/components/library/FolderTile.jsx]: Need to synchronize the API calls to delete a folder and add the restore notification.  First delete, then on a success callback, add the restore notification, then on a success callback, call `deletedFn`.  Albeit rare, with the current setup, you can 'undo' a delete before the folder is actually deleted.

In an ideal world, the undo would be robust enough to gracefully handle errors in one or both of the API calls.

File: app/js/components/folder/Add.jsx (hunk 4 above)
Comment (irizvifearless) [inline app/js/components/folder/Add.jsx]: the `onfetchLibraryCompleted` function is indented twice as far as it should and the F should be capitalized.

**Comment by irizvifearless** [comment]
Although it's hard to replicate, if I delete a folder and click 'undo' very quickly after, I can get the page to refresh with the restore notification, even though the folder has been restored.  Probably related to the fact that the API calls for delete and restore notification are not synchronous.

**Comment by blynch11p** [comment]
Good improvements to the Undo Delete Bookmark Folder feature.
Additionally, good update to prevent duplicating bookmarks unnecessarily when adding a shared folder.
👍

**Comment by irizvifearless** [comment]
Updates work as expected :+1:

**Pull request merged**